.github/workflows/tests.yml

```diff
@@ -201,12 +226,16 @@ jobs:
           path: bazel-testlogs
 
   multi_arch_support:
+    # i386 was removed on rules_apple 3.x.x - this test case needs reworking
+    # to exemplify fat binaries on the latest version
     name: Build iOS App for Multiple Architecture
     runs-on: macos-12
+    env:
+      USE_BAZEL_VERSION: 6.1.2
     steps:
       - uses: actions/checkout@v3
       - name: Preflight Env
-        run: .github/workflows/preflight_env.sh
+        run: .github/workflows/preflight_env.sh --no-bzlmod
```

Comment:
Bzlmod support was merged with 6.1.2, why should this be disabled here?

Reply:
[Please find in](https://github.com/bazel-ios/rules_ios/pull/773/files/9b0fe3fd7d2df44651e094308452e928aee83898#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR229) in the proceeding comment that it uses 6.1.2 and rules_apple 2 - where bzlmod is hardcoded to use 3.0.0.

Reply:
Gotcha so it doesnt work yet with rules_apple 3. I guess I got confused because we're essentially using Bazel `6.1.2` to mean rules_apple 2.

Not for this PR, but in the future I wonder if specifying these Bazel version + rules_apple versions wouldn't be better in a specific `WORKSPACE` for that specific test. Some repos do this with their integration tests and it allows them to test the different bazel version / rule versions they support without setting one version/feature for all their tests

Reply:
Nit: I'd still explicitly say in that comment why that implies `--no-bzlmod` is needed. The same way it's confusing during review it will be once we're changing this in the future.

Saw what @luispadron is talking about in other repos too and also like the idea. Wondering if we should create "Feature Request" templates for issues in `rules_ios` to track things like this 🤔

Reply:
Yeah it basically exists because people aren't using bzlmod on 5 and 6 so it's good to have some coverage of both ways to test this. Though, we could bump to a later version but needs more consideration.

Also, the way it's configured now get's us a larger matrix of bzlmod vs non bzlmod across a few versions. Though bzlmod support works on all the versions in the repo 5.4.1, 6.x.x we'd still like like the coverage to we don't break it - but probably not testing 100% of permutations